### ISSUE TITLE
Middleware edge functions do not support POST request with "Expect: 100-continue" HTTP header

### ISSUE THREAD

**Issue by Davy803** [zeit/now#8003]
# Steps to reproduce:

1. Add any valid `_middleware.ts` to a NextJS application.
2. Deploy to Vercel
3. Make a POST request to any url where that middleware will be run with: `Expect: 100-continue` as an HTTP header, and any non-empty `body` content.

# Result:
We get a `502 Bad Gateway` with the following response body:
```
An internal error occurred with Vercel.

INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED
```

In the function log, we get a message that the middleware was called, but the code does not execute (adding a console.info in the middleware does not show up in the function log)

# Expected result:
Middleware and page or api executes appropriately.

`Expect: 100-continue` is part of the HTTP 1.1 spec and should be implemented, or should be documented explicitly as not implemented.

**Comment by bmbshlly** [comment]
Why don't we support POST request in edge-functions?

**Comment by gabrielsestrem** [comment]
Hi, we've got many issues like this today in our application. I thought it was because some of my POST endpoint had query strings, bu even POST endpoints without any parameters got this error 502 INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED

**Comment by littleStudent** [comment]
👋 We are having the same issue since 12hours ago as well.
Seems like POST requests from some locations (confirmed via VPN for the UK) run into a 502 - INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED

From other parts of the world we do not see such a problem

**Comment by littleStudent** [comment]
Seems to be an issue with the latest nextjs version.
~~We pinned nextjs to 12.2.5 now and it works again.~~

I also dont think this is connected to the described topic from the initial author of this issue

**Comment by vandersoncostasantos** [comment]
Same issue here even with 12.2.5 version

**Comment by littleStudent** [comment]
Seems like we made a mistake and the middleware was not in use.
12.2.5 does have the same issue still.

Still havent figured out the root problem.

**Comment by bymoe** [comment]
Same here. Vercel is failing with `An internal error occurred with Vercel.` and `INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED` out of nowhere, since only ~15 hours.

**Comment by geoffw8** [comment]
+1 @bymoe

**Comment by seeARMS** [comment]
We're also encountering this issue, started within the last 24 hours. POST request hits error 502 with `INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED` and `An internal error occurred with Vercel`.

**Comment by brianarpie-forme** [comment]
+1

**Comment by fred-paragraph** [comment]
Happening here too.

**Comment by yuriykurylenko** [comment]
Here too.

**Comment by ethomson** [comment]
An update on this - we’ve identified and resolved the problem that many of you are reporting - where some requests are erroring with `INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED`. Please let me know if you’re still seeing this.

**Comment by gabrielsestrem** [comment]
Hi @ethomson . All my POST/PUT were erroring where we had query strings. I'll perform a test now and see if this was fixed. But just to be clear. I had to change many API methods in production to but our application live again. This error caused a huge outage in our application.

**Comment by gabrielsestrem** [comment]
Hi @ethomson , I can confirm the issue was fixed from our side. I'd like to mention something. This was reported on 21Jun. Then it happened again. Are you able to confirm that this bug it's nos gonna happen again?

**Comment by ethomson** [comment]
I’m not yet sure that these are the same issue, @gabrielsestrem - the original report is about `Expect: 100-continue`; todays issue happens to have the same error message (which is not uncommon).

But regardless of that, we’ll be adding additional testing here to ensure that we don’t regress today’s issue.

**Comment by Antonytm** [comment]
I have `[POST] [middleware: "_middleware"] /en/api/editing/render reason=INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED, status=502, user_error=false` 
Request ID: s5lwj-1667303455758-89e7a0c91110

The issue is reproducible only from some locations

**Comment by davidatfishtank** [comment]
I too  am seeing the exact same error @Antonytm 
[POST] [middleware: "_middleware"] /en/api/editing/render reason=INTERNAL_EDGE_FUNCTION_CONNECTION_CLOSED, status=502, user_error=false
Req ID: 8k6sr-1667402105158-e5c205d76798

**Comment by davidatfishtank** [comment]
Just thought I would tag @ethomson as you seem to be aware of previous related issues as it pertains to this problem.

**Comment by Antonytm** [comment]
@davidatfishtank 
I am waiting for Vercel's support response #00106486.

Due to the presence of "/en/api/editing/render", I guess you are also working with Sitecore JSS + Next.js.
What can probably work: 
1. Get rid of all existing middleware. (they may be not stable on some versions of Next.js)
2. Update to the latest Next.js version

**Comment by davidatfishtank** [comment]
@Antonytm you are correct.  Our middleware was pretty lightweight... what version of NextJs were you running?

**Comment by vieko** [comment]
I am seeing the same with a super light middleware on NextJS 12.2.4

**Comment by Antonytm** [comment]
@davidatfishtank  
We are on "next": "^12.1.5"
What about you?

**Comment by davidatfishtank** [comment]
@Antonytm 

sitecore-jss-nextjs": "^20.0.3"
"next": "^12.1.6"

**Comment by davidatfishtank** [comment]
@Antonytm just an FYI. Middleware eliminated, but issue still there... certainly something else afoot.

**Comment by vieko** [comment]
I can confirm this: the issue remains even without the middleware in place.

**Comment by davidatfishtank** [comment]
@Antonytm @vieko Are you still experiencing this issue?  At present, with our middleware back in, we are not. At this moment... 

But we are investigating migrating to NextJs 13 to see if it will help as 12.1 is clearly still beta territory.

**Comment by Antonytm** [comment]
@davidatfishtank 
We don't really need middleware for the short term. We disabled it and continue without it. Everything works.

BTW, Vercel support replied that there was an issue on their side. And it should work now. (But we didn't check)
JSS 20.0.3 did work with NextJs 13 for us. There were some edge cases when it failed. And we didn't want to be "testers of compatibility" of it on a real project. That is why we postponed NextJs update.
But it will interesting if you will manage it to work.

**Comment by vieko** [comment]
We brought back the middleware, working as expected. There was an issue on Vercel's end which has been fixed.
